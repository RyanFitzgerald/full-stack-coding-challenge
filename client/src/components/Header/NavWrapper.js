import styled from 'styled-components';

const NavWrapper = styled.div`
  float: right;
`;

export default NavWrapper;