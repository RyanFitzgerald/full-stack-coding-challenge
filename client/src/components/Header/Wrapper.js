import styled from 'styled-components';

const Wrapper = styled.header`
  background: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  padding: 15px 0;
`;

export default Wrapper;