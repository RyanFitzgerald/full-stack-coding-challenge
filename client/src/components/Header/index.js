import React from 'react';
import { Link } from 'react-router-dom';

// Import static files
import Logo from '../../static/logo.png';

// Import components
import Wrapper from './Wrapper';
import LogoWrapper from './LogoWrapper';
import NavWrapper from './NavWrapper';
import Container from '../Container';
import Nav from '../Nav';

function Header() {
  return (
    <Wrapper>
      <Container>
        <LogoWrapper>
          <Link to="/">
            <img src={Logo} alt="Address Lookup"/>
          </Link>
        </LogoWrapper>
        <NavWrapper>
          <Nav/>
        </NavWrapper>
      </Container>
    </Wrapper>
  );
}

export default Header;