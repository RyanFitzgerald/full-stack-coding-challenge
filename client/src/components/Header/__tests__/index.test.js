import React from 'react';
import { StaticRouter } from 'react-router-dom';

import Header from '../index';

describe('<Header />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <StaticRouter location="/">
        <Header/>
      </StaticRouter>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should contain navigation', () => {
    const renderedComponent = mount(
      <StaticRouter location="/">
        <Header/>
      </StaticRouter>
    );
    
    expect(renderedComponent.find('nav')).toHaveLength(1);
  });
});