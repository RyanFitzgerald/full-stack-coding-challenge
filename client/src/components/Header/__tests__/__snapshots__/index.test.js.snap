// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
.c0 {
  background: #fff;
  border-bottom: 1px solid #e6e6e6;
  padding: 15px 0;
}

.c2 {
  float: left;
}

.c3 {
  float: right;
}

.c1 {
  margin: 0 auto;
  max-width: 960px;
  padding: 15px;
}

.c1:after {
  content: '';
  display: table;
  clear: both;
}

.c4 {
  border-radius: 4px;
  color: #696d80;
  display: inline-block;
  font-weight: 700;
  padding: 15px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.25s ease all;
  transition: 0.25s ease all;
}

.c4:hover {
  background: #f2f2f2;
  color: #3498db;
}

<header
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Address Lookup"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="c3"
    >
      <nav>
        <a
          className="c4"
          href="/"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="c4"
          href="/add"
          onClick={[Function]}
        >
          Add
        </a>
      </nav>
    </div>
  </div>
</header>
`;
